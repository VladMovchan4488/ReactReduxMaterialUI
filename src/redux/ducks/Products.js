import { takeLatest } from "redux-saga/effects";
import { put, call } from "redux-saga/effects";

// Actions
export const GET_PRODUCTS = "GET_PRODUCTS";
export const GET_PRODUCTS_API = "GET_PRODUCTS_API";
export const SHOW_TABLE = "SHOW_TABLE";

// Default State
export const initialState = {
  products: [],
  isLoading: [1]
};

// Dispatch
export const getProductsApi = () => ({ type: GET_PRODUCTS_API });
export const getShow = () => ({ type: SHOW_TABLE });

// Reducer
const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_PRODUCTS:
      return {
        ...state,
        products: action.data.products
      };
    case SHOW_TABLE:
      return {
        ...state,
        isLoading: [...state.isLoading, 2]
      };
    default:
      return state;
  }
};

// REDUX-SAGA
export function* getProducts() {
  const url = "/products.json";
  try {
    const data = yield call(() => {
      return fetch(url).then(res => res.json());
    });
    yield put({ type: GET_PRODUCTS, data });
  } catch {
    console.error("error");
  }
}

export function* parsing() {
  yield takeLatest(GET_PRODUCTS_API, getProducts);
}

export default reducer;
