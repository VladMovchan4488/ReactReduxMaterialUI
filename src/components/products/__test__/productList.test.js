import React from "react";
import Products from "../productList";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

it("Snaphot List", () => {
  const props = {
    products: [
      {
        bsr_category: "Home & Kitchen",
        price: 15.79,
        stars: 4.6,
        name: "A3 Magnetic Dry Wipe Weekly Planner Whiteboard for Fridge"
      }
    ],
    isLoading: [1]
  };

  const wrapper = shallow(<Products.WrappedComponent {...props} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
