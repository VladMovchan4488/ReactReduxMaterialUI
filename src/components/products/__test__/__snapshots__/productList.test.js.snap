// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snaphot List 1`] = `
<div
  className="render"
>
  <Product
    products={
      Array [
        Object {
          "bsr_category": "Home & Kitchen",
          "name": "A3 Magnetic Dry Wipe Weekly Planner Whiteboard for Fridge",
          "price": 15.79,
          "stars": 4.6,
        },
      ]
    }
  />
  <button
    className="btn btn-secondary addAction"
    onClick={[Function]}
    type="button"
  >
    Добавить ещё одну таблицу
  </button>
</div>
`;
